api/v1alpha1/shared_types.go

```diff
@@ -676,21 +691,39 @@ type CustomResponse struct {
 // +kubebuilder:validation:Enum=Inline;ValueRef
 type ResponseValueType string
 
+const (
+	// ResponseValueTypeInline defines the "Inline" response body type.
+	ResponseValueTypeInline ResponseValueType = "Inline"
+
+	// ResponseValueTypeValueRef defines the "ValueRef" response body type.
+	ResponseValueTypeValueRef ResponseValueType = "ValueRef"
+)
+
 // CustomResponseBody
+// +kubebuilder:validation:XValidation:message="inline must be set for type Inline",rule="(!has(self.type) || self.type == 'Inline')? has(self.inline) : true"
+// +kubebuilder:validation:XValidation:message="valueRef must be set for type ValueRef",rule="(has(self.type) && self.type == 'ValueRef')? has(self.valueRef) : true"
+// +kubebuilder:validation:XValidation:message="only ConfigMap is supported for ValueRef",rule="has(self.valueRef) ? self.valueRef.kind == 'ConfigMap' : true"
 type CustomResponseBody struct {
 	// Type is the type of method to use to read the body value.
+	// Valid values are Inline and ValueRef, default is Inline.
 	//
+	// +kubebuilder:default=Inline
+	// +kubebuilder:validation:Enum=Inline;ValueRef
 	// +unionDiscriminator
 	Type *ResponseValueType `json:"type"`
 
 	// Inline contains the value as an inline string.
 	//
 	// +optional
 	Inline *string `json:"inline,omitempty"`
+
 	// ValueRef contains the contents of the body
 	// specified as a local object reference.
 	// Only a reference to ConfigMap is supported.
 	//
+	// The contents of the ConfigMap must have a key-value pair where
+	// the key is `response.body` and the value is the body content.
```

Comment:
do we need to enforce the key name here ? Any key name (like file name) should be fine imo, EG will look into `value[0]` for the data

Reply:
Explicitly asking for a fixed name creates a better UI, but we can fall back to value[0] if this key doesn't exit. WDYT?

Reply:
@envoyproxy/gateway-maintainers Could you please vote for the name of the response body key in the ConfigMap?
* response.body
* body 
* Or other names that you suggest?

Reply:
+1 for `request.body`